File: python/trustlayer/modules/agent.py
"""AgentShield — post an agent id and evaluate. The id is not authorization."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any, Optional

if TYPE_CHECKING:
    from ..session import TrustSession


class AgentShield:
    def __init__(self, session: "TrustSession") -> None:
        self._session = session

    def evaluate(self, agent_id: str, owner_org: Optional[str] = None) -> Any:
        self._session.track_event(
            "custom",
            {"agent_id": agent_id, "owner_org": owner_org},
        )
        return self._session.evaluate(modules=["agent", "bot", "spam"])

File: src/modules/agent.ts
import { TrustSession } from "../session";
import { EvaluationResponse } from "../api/types";

export interface AgentShieldResult {
  trustScore: number;
  behaviorRisk: number;
  signals: string[];
  recommendation: EvaluationResponse["recommendation"];
}

/**
 * AgentShield posts an agent id and evaluates the agent module.
 * A self-declared id is not authorization.
 */
export class AgentShield {
  constructor(private readonly session: TrustSession) {}

  async evaluate(agentId: string, ownerOrg?: string): Promise<AgentShieldResult> {
    await this.session.trackEvent("custom", {
      agent_id: agentId,
      owner_org: ownerOrg,
    });
    const evaluation = await this.session.evaluate(["agent", "bot", "spam"]);
    const agent = evaluation.modules?.agent;
    return {
      trustScore: agent?.trust_score ?? 0,
      behaviorRisk: agent?.behavior_risk ?? 0,
      signals: agent?.signals ?? [],
      recommendation: evaluation.recommendation,
    };
  }
}
